history.py

```diff
@@ -0,0 +1,36 @@
+from catalog import models as cmod
+
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        recent_list = request.session.get('last_five', [])
+        request.last_five = []
+
```

Comment:
While good practice for readability, excessive blank lines of code can potentially slow your system down. Instead, try adding #comments after each new 'section' of code in your def.